aten/src/ATen/core/function_schema_inl.h

```diff
@@ -74,6 +103,82 @@ inline std::string FunctionSchema::formatTypeMismatchMsg(
       *this);
 }
 
+inline bool FunctionSchema::isBackwardCompatibleWith(
+    const FunctionSchema& old,
+    std::ostream* why_not) const {
+  if (!(name() == old.name()
+        && overload_name() == old.overload_name()
+        // we are conservative on is_vararg and is_varret,
+        // since they are only used by internal operators
+        && is_vararg() == old.is_vararg()
+        && is_varret() == old.is_varret()
+        && returns().size() == old.returns().size()
+        && arguments().size() >= old.arguments().size())) {
+    return false;
+  }
+  for (size_t i = 0; i < returns().size(); ++i) {
+    if (!returns().at(i).isBackwardCompatibleWith(
```

Comment:
Just swap the arguments here and put the contravariant comment here.

Reply:
done